tests/unit/acquisition/multi_objective/test_pareto.py

```diff
@@ -104,3 +105,74 @@ def test_get_reference_point_extract_based_on_pareto_front(
     observations: tf.Tensor, expected: tf.Tensor
 ) -> None:
     tf.debugging.assert_equal(get_reference_point(observations), expected)
+
+
+@pytest.mark.qhsri
+def test_pareto_sample_diverse_subset_raises_too_large_sample_size() -> None:
+    observations = tf.constant([[1.0, -1.0], [-1.0, 1.0]])
+    pareto_set = Pareto(observations)
+    with pytest.raises(ValueError):
+        pareto_set.sample_diverse_subset(3, allow_repeats=False)
+
+
+@pytest.mark.qhsri
+def test_pareto_sample_diverse_subset_raises_zero_range() -> None:
+    observations = tf.constant([[1.0, 1.0], [1.0, 1.0]])
+    pareto_set = Pareto(observations)
+    with pytest.raises(ValueError):
+        pareto_set.sample_diverse_subset(1, bounds_min_delta=0.0)
+
+
+@pytest.mark.qhsri
+def test_pareto_sample_diverse_subset_get_bounds() -> None:
+    observations = tf.constant([[1.0, -1.0], [-1.0, 1.0]])
+    pareto_set = Pareto(observations)
+    lower_bounds, reference_point = pareto_set._get_bounds(delta_scaling_factor=0.2, min_delta=1e-9)
+    expected_lower_bounds = tf.constant([-1.4, -1.4])
+    expected_reference_point = tf.constant([1.4, 1.4])
+    npt.assert_allclose(expected_lower_bounds, lower_bounds)
+    npt.assert_allclose(expected_reference_point, reference_point)
+
+
+@pytest.mark.qhsri
+def test_pareto_sample_diverse_subset_calculate_p() -> None:
+    observations = tf.constant([[1.0, -1.0], [-1.0, 1.0]])
+    lower_bound = tf.constant([-2.0, -2.0])
+    reference_point = tf.constant([2.0, 2.0])
+    pareto_set = Pareto(observations)
+    output = pareto_set._calculate_p_matrix(lower_bound, reference_point)
+    expected_output = tf.constant([[3 / 16, 1 / 16], [1 / 16, 3 / 16]])
+    npt.assert_array_equal(expected_output, output)
+
+
+@pytest.mark.qhsri
+def test_pareto_sample_diverse_subset_choose_batch_no_repeats() -> None:
+    observations = tf.constant([[2.0, -2.0], [1.0, -1.0], [0.0, 0.0], [-1.0, 1.0], [-2.0, 2.0]])
+    x_star = tf.constant([[0.15], [0.25], [0.2], [0.3], [0.1]])
+    pareto_set = Pareto(observations)
+    sample, sample_ids = pareto_set._choose_batch_no_repeats(x_star, sample_size=2)
+    expected_sample = tf.constant([[-1.0, 1.0], [1.0, -1.0]])
+    expected_sample_ids = tf.constant([3, 1])
+    npt.assert_array_equal(expected_sample, sample)
+    npt.assert_array_equal(expected_sample_ids, sample_ids)
+
+
+@pytest.mark.parametrize(
+    "x_star,expected_ids",
```

Comment:
do you want to add a test where choosing 3 points from a batch of 3 returns the same front?